src/orgwarden/repo_crawler.py

```diff
@@ -1,37 +1,38 @@
-import requests
+import json
+import subprocess
 from orgwarden.repository import Repository
 
 
-def fetch_org_repos(org_name: str) -> list[Repository]:
+class APIError(Exception):
+    def __init__(self, message: str):
+        super().__init__(message)
+
+
+def fetch_org_repos(org_name: str, hostname: str) -> list[Repository]:
     """
     Returns a `Repository` list containing the specified organization's public, non-forked repositories.
+    Raises an `APIError` if an error occurs while fetching repositories.
     """
 
     org_repo_entries: list[dict] = []  # unfiltered api response
-    page_num = 1
-    while True:
-        response = requests.get(
-            url=f"https://api.github.com/orgs/{org_name}/repos",
-            headers={"Accept": "application/vnd.github+json"},
-            params={
-                "per_page": 100,  # max repos per page
-                "page": page_num,
-            },
-        )
-        if not response or response.status_code != 200:
-            raise requests.HTTPError(
-                f"Error fetching repos for organization: {org_name}", response=response
-            )
 
-        json_data: list[dict] = response.json()
-        if not json_data:
-            break  # No more repos
-        org_repo_entries += json_data
-        page_num += 1  # Go to the next page
+    res = subprocess.run(
+        f"gh api orgs/{org_name}/repos --hostname {hostname} --paginate",
```

Comment:
Taking a dependency on `gh` here makes this code easier to write, but we should now introduce functionality to check that `gh` is installed locally and produce an error message if it is not. We should also ensure that we have tests that exercise the code path when `gh` is not installed.